### PR TITLE
Support of included files list

### PR DESCRIPTION
**Pull request by alinadubatovka**
TexCompleter processes passed lists of included files and all project bibliography entries to range autocompletions generated from included biblio files higher than others

### CODE REVIEW

File: lib/ace/ext/papeeria/tex_completer.coffee [hunk 1]
```diff
@@ -252,6 +255,7 @@ define((require, exports, module) ->
       constructor: ->
         @refCache = new CompletionsCache(getJsonFromUrl, processReferenceJson)
         @citeCache = new CompletionsCache(getJsonFromUrl, processCitationJson)
+        @filesCache = new CompletionsCache(getJsonFromUrl, processFilesJson)
```
Comment (dbarashev) [inline lib/ace/ext/papeeria/tex_completer.coffee]: If `processFilesJson` is not supposed to do anything other than returning its argument, you may want to replace it with `(json) -> json`

File: lib/ace/ext/papeeria/tex_completer.coffee [hunk 2]
```diff
@@ -314,7 +320,16 @@ define((require, exports, module) ->
           return
 
         if LatexParsingContext.isType(token, "cite")
-          if @citationsUrl? then @citeCache.getReferences(@citationsUrl, callback)
+          if @citationsUrl? then @citeCache.getReferences(@citationsUrl, (err, cites) =>
```
Comment (dbarashev) [inline lib/ace/ext/papeeria/tex_completer.coffee]: Can you write a comment for these tricks? Also, if we have no `filesUrl` for whatever reason, we won't show any completions at all, right?

File: lib/ace/ext/papeeria/tex_completer.coffee (hunk 1 above)
Comment (alinadubatovka) [inline lib/ace/ext/papeeria/tex_completer.coffee]: Done

File: lib/ace/ext/papeeria/tex_completer.coffee (hunk 2 above)
Comment (alinadubatovka) [inline lib/ace/ext/papeeria/tex_completer.coffee]: Apparently yes. Fixed

File: lib/ace/ext/papeeria/tex_completer.coffee [hunk 3]
```diff
@@ -314,7 +317,26 @@ define((require, exports, module) ->
...
+            if @filesUrl?
+              # After we get a list of cite autocompletion we want to extract entries from already included biblio files
+              # Let's retrieve a list of files included to the current target from the filesUrl
+              @filesCache.getReferences(@filesUrl, (err, files) =>
+                # Once we have a list of files included to the compilation we can find citation keys that
+                # are (not) included and treat them differently
+                cites.map((cite) =>
+                    if !(cite.meta in files)
```
Comment (dbarashev) [inline lib/ace/ext/papeeria/tex_completer.coffee]: I am curious if it is possible at all. When we build cites.json, we take the list of bib files from aux file produces by tex, which means that all files mentioned in cites belong to the current target by definition. Or do you plan to change this?

**Pull request merged**